### PR TITLE
Fix problem with multiple years and output_format

### PR DESCRIPTION
**Pull request by agila5**
Tomorrow morning I will check this again but it should work. 

``` r
library(stats19)
#> Data provided under OGL v3.0. Cite the source and link to:
#> www.nationalarchives.gov.uk/doc/open-government-licence/version/3/
get_stats19(c(2017, 2018), silent = TRUE)
#> date and time columns present, creating formatted datetime column
#> date and time columns present, creating formatted datetime column
#> # A tibble: 252,617 x 33
#>    accident_index location_eastin… location_northi… longitude latitude
#>    <chr>                     <int>            <int>     <dbl>    <dbl>
#>  1 2017010001708            532920           196330   -0.0801     51.7
#>  2 2017010009342            526790           181970   -0.174      51.5
#>  3 2017010009344            535200           181260   -0.0530     51.5
#>  4 2017010009348            534340           193560   -0.0607     51.6
#>  5 2017010009350            533680           187820   -0.0724     51.6
#>  6 2017010009351            514510           172370   -0.354      51.4
#>  7 2017010009353            508640           181870   -0.435      51.5
#>  8 2017010009354            527880           181950   -0.158      51.5
#>  9 2017010009357            520940           192820   -0.254      51.6
#> 10 2017010009358            531430           178450   -0.108      51.5
#> # … with 252,607 more rows, and 28 more variables: police_force <chr>,
#> #   accident_severity <chr>, number_of_vehicles <int>,
#> #   number_of_casualties <int>, date <date>, day_of_week <chr>, time <chr>,
#> #   local_authority_district <chr>, local_authority_highway <chr>,
#> #   first_road_class <chr>, first_road_number <int>, road_type <chr>,
#> #   speed_limit <int>, junction_detail <chr>, junction_control <chr>,
#> #   second_road_class <chr>, second_road_number <int>,
#> #   pedestrian_crossing_human_control <chr>,
#> #   pedestrian_crossing_physical_facilities <chr>, light_conditions <chr>,
#> #   weather_conditions <chr>, road_surface_conditions <chr>,
#> #   special_conditions_at_site <chr>, carriageway_hazards <chr>,
#> #   urban_or_rural_area <chr>,
#> #   did_police_officer_attend_scene_of_accident <int>,
#> #   lsoa_of_accident_location <chr>, datetime <dttm>
get_stats19(c(2017, 2018), silent = TRUE, output_format = "sf")
#> date and time columns present, creating formatted datetime column
#> 19 rows removed with no coordinates
#> date and time columns present, creating formatted datetime column
#> 55 rows removed with no coordinates
#> Simple feature collection with 252543 features and 31 fields
#> geometry type:  POINT
#> dimension:      XY
#> bbox:           xmin: 73639 ymin: 10235 xmax: 655391 ymax: 1209512
#> epsg (SRID):    27700
#> proj4string:    +proj=tmerc +lat_0=49 +lon_0=-2 +k=0.9996012717 +x_0=400000 +y_0=-100000 +ellps=airy +units=m +no_defs
#> # A tibble: 252,543 x 32
#>    accident_index longitude latitude police_force accident_severi…
#>    <chr>              <dbl>    <dbl> <chr>        <chr>           
#>  1 2017010001708    -0.0801     51.7 Metropolita… Fatal           
#>  2 2017010009342    -0.174      51.5 Metropolita… Slight          
#>  3 2017010009344    -0.0530     51.5 Metropolita… Slight          
#>  4 2017010009348    -0.0607     51.6 Metropolita… Slight          
#>  5 2017010009350    -0.0724     51.6 Metropolita… Serious         
#>  6 2017010009351    -0.354      51.4 Metropolita… Slight          
#>  7 2017010009353    -0.435      51.5 Metropolita… Slight          
#>  8 2017010009354    -0.158      51.5 Metropolita… Slight          
#>  9 2017010009357    -0.254      51.6 Metropolita… Serious         
#> 10 2017010009358    -0.108      51.5 Metropolita… Serious         
#> # … with 252,533 more rows, and 27 more variables: number_of_vehicles <int>,
#> #   number_of_casualties <int>, date <date>, day_of_week <chr>, time <chr>,
#> #   local_authority_district <chr>, local_authority_highway <chr>,
#> #   first_road_class <chr>, first_road_number <int>, road_type <chr>,
#> #   speed_limit <int>, junction_detail <chr>, junction_control <chr>,
#> #   second_road_class <chr>, second_road_number <int>,
#> #   pedestrian_crossing_human_control <chr>,
#> #   pedestrian_crossing_physical_facilities <chr>, light_conditions <chr>,
#> #   weather_conditions <chr>, road_surface_conditions <chr>,
#> #   special_conditions_at_site <chr>, carriageway_hazards <chr>,
#> #   urban_or_rural_area <chr>,
#> #   did_police_officer_attend_scene_of_accident <int>,
#> #   lsoa_of_accident_location <chr>, datetime <dttm>, geometry <POINT [m]>
get_stats19(c(2017, 2018), silent = TRUE, output_format = "ppp")
#> date and time columns present, creating formatted datetime column
#> Registered S3 method overwritten by 'spatstat':
#>   method     from
#>   print.boxx cli
#> 19 rows removed with no coordinates
#> Warning: some mark values are NA in the point pattern x
#> date and time columns present, creating formatted datetime column
#> 55 rows removed with no coordinates
#> Warning: some mark values are NA in the point pattern x

#> Warning: some mark values are NA in the point pattern x

#> Warning: some mark values are NA in the point pattern x
#> Marked planar point pattern: 252543 points
#> Mark variables: 
#>    accident_index longitude latitude police_force accident_severity 
#> number_of_vehicles number_of_casualties date day_of_week time 
#> local_authority_district local_authority_highway first_road_class 
#> first_road_number road_type speed_limit junction_detail junction_control 
#> second_road_class second_road_number pedestrian_crossing_human_control 
#> pedestrian_crossing_physical_facilities light_conditions weather_conditions 
#> road_surface_conditions special_conditions_at_site carriageway_hazards 
#> urban_or_rural_area did_police_officer_attend_scene_of_accident 
#> lsoa_of_accident_location datetime
#> window: rectangle = [64950, 655391] x [10235, 1209512] units
```

<sup>Created on 2020-01-30 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

I don't like the testing component since it downloads the same data several times. Suggestions?

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Great to see these latest changes, many thanks @agila5. One more thought: is it worth adding `data.frame` as an output option and coercing the default `tibble` with `as.data.frame()` if people select the `data.frame` option? 

Reasoning, not everyone likes tibble output and it's an easy addition.

**Comment by agila5** [comment]
Done! It doesn't feel 100% natural to read stats19 data as tbl data with readr::read_csv and then transform the tbl data into a data.frame but that's unimportant IMO.

File: tests/testthat/test-get.R [hunk 1]
```diff
@@ -37,3 +39,13 @@ test_that("get_stats19 multiple years", {
 
   expect_equal(nrow(t), sum(nrow(t09), nrow(t15), nrow(t16)))
 })
+
+test_that("get_stats19 works with multiple years and formats", {
+  t1718 <- get_stats19(c(2017, 2018), output_format = "sf")
```
Comment (Robinlovelace) [inline tests/testthat/test-get.R]: Not in `stats19` style but happy to tidy-up after merge : )

**Comment by Robinlovelace** [comment]
FYI @wengraf and @layik here's a demo of the new functionaility:

``` r
remotes::install_github("ropensci/stats19")
#> Skipping install of 'stats19' from a github remote, the SHA1 (579f5992) has not changed since last install.
#>   Use `force = TRUE` to force installation
library(stats19)
#> Data provided under OGL v3.0. Cite the source and link to:
#> www.nationalarchives.gov.uk/doc/open-government-licence/version/3/
a = get_stats19(2018, "ac")
#> Files identified: dftRoadSafetyData_Accidents_2018.csv
#>    http://data.dft.gov.uk.s3.amazonaws.com/road-accidents-safety-data/dftRoadSafetyData_Accidents_2018.csv
#> Data already exists in data_dir, not downloading
#> Data saved at ~/stats19-data/dftRoadSafetyData_Accidents_2018.csv
#> Reading in:
#> ~/stats19-data/dftRoadSafetyData_Accidents_2018.csv
#> date and time columns present, creating formatted datetime column
adf = get_stats19(2018, "ac", output_format = "data.frame")
#> Files identified: dftRoadSafetyData_Accidents_2018.csv
#>    http://data.dft.gov.uk.s3.amazonaws.com/road-accidents-safety-data/dftRoadSafetyData_Accidents_2018.csv
#> Data already exists in data_dir, not downloading
#> Data saved at ~/stats19-data/dftRoadSafetyData_Accidents_2018.csv
#> Reading in:
#> ~/stats19-data/dftRoadSafetyData_Accidents_2018.csv
#> date and time columns present, creating formatted datetime column
asf = get_stats19(2018, "ac", output_format = "sf")
#> Files identified: dftRoadSafetyData_Accidents_2018.csv
#>    http://data.dft.gov.uk.s3.amazonaws.com/road-accidents-safety-data/dftRoadSafetyData_Accidents_2018.csv
#> Data already exists in data_dir, not downloading
#> Data saved at ~/stats19-data/dftRoadSafetyData_Accidents_2018.csv
#> Reading in:
#> ~/stats19-data/dftRoadSafetyData_Accidents_2018.csv
#> date and time columns present, creating formatted datetime column
#> 55 rows removed with no coordinates
appp = get_stats19(2018, "ac", output_format = "ppp")
#> Files identified: dftRoadSafetyData_Accidents_2018.csv
#>    http://data.dft.gov.uk.s3.amazonaws.com/road-accidents-safety-data/dftRoadSafetyData_Accidents_2018.csv
#> Data already exists in data_dir, not downloading
#> Data saved at ~/stats19-data/dftRoadSafetyData_Accidents_2018.csv
#> Reading in:
#> ~/stats19-data/dftRoadSafetyData_Accidents_2018.csv
#> date and time columns present, creating formatted datetime column
#> 55 rows removed with no coordinates
#> Warning: some mark values are NA in the point pattern x

class(a)
#> [1] "spec_tbl_df" "tbl_df"      "tbl"         "data.frame"
class(adf)
#> [1] "data.frame"
class(asf)
#> [1] "sf"         "tbl_df"     "tbl"        "data.frame"
class(appp)
#> [1] "ppp"

head(a)
#> Registered S3 method overwritten by 'cli':
#>   method     from    
#>   print.boxx spatstat
#> # A tibble: 6 x 33
#>   accident_index location_eastin… location_northi… longitude latitude
#>   <chr>                     <int>            <int>     <dbl>    <dbl>
#> 1 2018010080971            529150           182270   -0.140      51.5
#> 2 2018010080973            542020           184290    0.0465     51.5
#> 3 2018010080974            531720           182910   -0.102      51.5
#> 4 2018010080981            541450           183220    0.0378     51.5
#> 5 2018010080982            543580           176500    0.0658     51.5
#> 6 2018010080983            526060           194910   -0.180      51.6
#> # … with 28 more variables: police_force <chr>, accident_severity <chr>,
#> #   number_of_vehicles <int>, number_of_casualties <int>, date <date>,
#> #   day_of_week <chr>, time <chr>, local_authority_district <chr>,
#> #   local_authority_highway <chr>, first_road_class <chr>,
#> #   first_road_number <int>, road_type <chr>, speed_limit <int>,
#> #   junction_detail <chr>, junction_control <chr>, second_road_class <chr>,
#> #   second_road_number <int>, pedestrian_crossing_human_control <chr>,
#> #   pedestrian_crossing_physical_facilities <chr>, light_conditions <chr>,
#> #   weather_conditions <chr>, road_surface_conditions <chr>,
#> #   special_conditions_at_site <chr>, carriageway_hazards <chr>,
#> #   urban_or_rural_area <chr>,
#> #   did_police_officer_attend_scene_of_accident <int>,
#> #   lsoa_of_accident_location <chr>, datetime <dttm>
head(adf)
#>   accident_index location_easting_osgr location_northing_osgr longitude
#> 1  2018010080971                529150                 182270 -0.139737
#> 2  2018010080973                542020                 184290  0.046471
#> 3  2018010080974                531720                 182910 -0.102474
#> 4  2018010080981                541450                 183220  0.037828
#> 5  2018010080982                543580                 176500  0.065781
#> 6  2018010080983                526060                 194910 -0.179720
#>   latitude        police_force accident_severity number_of_vehicles
#> 1 51.52459 Metropolitan Police            Slight                  2
#> 2 51.53965 Metropolitan Police            Slight                  1
#> 3 51.52975 Metropolitan Police            Slight                  2
#> 4 51.53018 Metropolitan Police           Serious                  2
#> 5 51.46926 Metropolitan Police           Serious                  2
#> 6 51.63888 Metropolitan Police            Slight                  2
#>   number_of_casualties       date day_of_week  time local_authority_district
#> 1                    2 2018-01-01      Monday 01:30                   Camden
#> 2                    1 2018-01-01      Monday 00:50                   Newham
#> 3                    1 2018-01-01      Monday 00:45                Islington
#> 4                    1 2018-01-01      Monday 03:00                   Newham
#> 5                    2 2018-01-01      Monday 02:20                Greenwich
#> 6                    4 2018-01-01      Monday 01:52                   Barnet
#>   local_authority_highway first_road_class first_road_number          road_type
#> 1                  Camden                A               501   Dual carriageway
#> 2                  Newham                B               165 Single carriageway
#> 3               Islington                A                 1 Single carriageway
#> 4                  Newham                B               167   Dual carriageway
#> 5               Greenwich                A               207 Single carriageway
#> 6                  Barnet                A              1000 Single carriageway
#>   speed_limit                     junction_detail             junction_control
#> 1          30 Not at junction or within 20 metres Data missing or out of range
#> 2          30                     Mini-roundabout     Give way or uncontrolled
#> 3          20                          Crossroads     Give way or uncontrolled
#> 4          30   More than 4 arms (not roundabout)          Auto traffic signal
#> 5          30 Not at junction or within 20 metres Data missing or out of range
#> 6          30 Not at junction or within 20 metres Data missing or out of range
#>   second_road_class second_road_number pedestrian_crossing_human_control
#> 1              <NA>                  0             None within 50 metres
#> 2      Unclassified                  0             None within 50 metres
#> 3                 C                  0             None within 50 metres
#> 4                 A                124             None within 50 metres
#> 5              <NA>                  0             None within 50 metres
#> 6              <NA>                  0             None within 50 metres
#>            pedestrian_crossing_physical_facilities      light_conditions
#> 1 No physical crossing facilities within 50 metres Darkness - lights lit
#> 2 No physical crossing facilities within 50 metres Darkness - lights lit
#> 3      Pedestrian phase at traffic signal junction Darkness - lights lit
#> 4      Pedestrian phase at traffic signal junction Darkness - lights lit
#> 5 No physical crossing facilities within 50 metres Darkness - lights lit
#> 6 No physical crossing facilities within 50 metres Darkness - lights lit
#>      weather_conditions road_surface_conditions special_conditions_at_site
#> 1    Fine no high winds                     Dry                       None
#> 2    Fine no high winds                     Dry                       None
#> 3    Fine no high winds                     Dry                       None
#> 4 Raining no high winds             Wet or damp                       None
#> 5    Fine no high winds             Wet or damp                       None
#> 6  Raining + high winds             Wet or damp                       None
#>   carriageway_hazards urban_or_rural_area
#> 1                None               Urban
#> 2                None               Urban
#> 3                None               Urban
#> 4                None               Urban
#> 5                None               Urban
#> 6                None               Urban
#>   did_police_officer_attend_scene_of_accident lsoa_of_accident_location
#> 1                                           1                 E01000854
#> 2                                           1                 E01003531
#> 3                                           1                 E01002723
#> 4                                           1                 E01003492
#> 5                                           1                 E01001682
#> 6                                           1                 E01000271
#>              datetime
#> 1 2018-01-01 01:30:00
#> 2 2018-01-01 00:50:00
#> 3 2018-01-01 00:45:00
#> 4 2018-01-01 03:00:00
#> 5 2018-01-01 02:20:00
#> 6 2018-01-01 01:52:00
head(asf)
#> Simple feature collection with 6 features and 31 fields
#> geometry type:  POINT
#> dimension:      XY
#> bbox:           xmin: 526060 ymin: 176500 xmax: 543580 ymax: 194910
#> epsg (SRID):    27700
#> proj4string:    +proj=tmerc +lat_0=49 +lon_0=-2 +k=0.9996012717 +x_0=400000 +y_0=-100000 +ellps=airy +units=m +no_defs
#> # A tibble: 6 x 32
#>   accident_index longitude latitude police_force accident_severi…
#>   <chr>              <dbl>    <dbl> <chr>        <chr>           
#> 1 2018010080971    -0.140      51.5 Metropolita… Slight          
#> 2 2018010080973     0.0465     51.5 Metropolita… Slight          
#> 3 2018010080974    -0.102      51.5 Metropolita… Slight          
#> 4 2018010080981     0.0378     51.5 Metropolita… Serious         
#> 5 2018010080982     0.0658     51.5 Metropolita… Serious         
#> 6 2018010080983    -0.180      51.6 Metropolita… Slight          
#> # … with 27 more variables: number_of_vehicles <int>,
#> #   number_of_casualties <int>, date <date>, day_of_week <chr>, time <chr>,
#> #   local_authority_district <chr>, local_authority_highway <chr>,
#> #   first_road_class <chr>, first_road_number <int>, road_type <chr>,
#> #   speed_limit <int>, junction_detail <chr>, junction_control <chr>,
#> #   second_road_class <chr>, second_road_number <int>,
#> #   pedestrian_crossing_human_control <chr>,
#> #   pedestrian_crossing_physical_facilities <chr>, light_conditions <chr>,
#> #   weather_conditions <chr>, road_surface_conditions <chr>,
#> #   special_conditions_at_site <chr>, carriageway_hazards <chr>,
#> #   urban_or_rural_area <chr>,
#> #   did_police_officer_attend_scene_of_accident <int>,
#> #   lsoa_of_accident_location <chr>, datetime <dttm>, geometry <POINT [m]>
head(appp)
#> Warning: some mark values are NA in the point pattern x
#> Marked planar point pattern: 6 points
#> Mark variables: 
#>    accident_index longitude latitude police_force accident_severity 
#> number_of_vehicles number_of_casualties date day_of_week time 
#> local_authority_district local_authority_highway first_road_class 
#> first_road_number road_type speed_limit junction_detail junction_control 
#> second_road_class second_road_number pedestrian_crossing_human_control 
#> pedestrian_crossing_physical_facilities light_conditions weather_conditions 
#> road_surface_conditions special_conditions_at_site carriageway_hazards 
#> urban_or_rural_area did_police_officer_attend_scene_of_accident 
#> lsoa_of_accident_location datetime
#> window: rectangle = [64950, 655391] x [10235, 1209512] units
```

<sup>Created on 2020-02-03 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Pull request merged**